### ISSUE TITLE
GrafanaUI: Setting required=true on InlineField does not work

### ISSUE THREAD

**Issue by JoaoSilvaGrafana** [grafana/grafana#54848]
**What happened**:

If the `InlineField` component has `required` set to `true`, that does not do anything.
<img width="426" alt="Screenshot 2022-09-07 at 14 20 50" src="https://user-images.githubusercontent.com/100691367/188888682-4a378169-aa9b-4c72-b72a-8c99c1c54bc3.png">

**What you expected to happen**:

A little "*" appears next to the label, just like for the `Field` component
<img width="464" alt="Screenshot 2022-09-07 at 14 21 10" src="https://user-images.githubusercontent.com/100691367/188888766-dd96eaa5-c200-46db-9eea-e81b842ac279.png">

**How to reproduce it (as minimally and precisely as possible)**:

Once https://github.com/grafana/grafana/pull/54674 is merged, run storybook locally and try to turn on the `required` control.

**Environment**:
- Grafana version: 9.1.3
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

**Comment by Wafa9** [comment]
HI @gefgu  I would like to work on this issue, I am looking for a good first issue so can I take this.

**Comment by gefgu** [comment]
@Wafa9 I've already made the pull request with the required changes. You can try to find another issue. Grafana always has good open-source issues to work on.
